### ISSUE TITLE
libunwind: cmake build doesn't define it will use assembler files

### ISSUE THREAD

**Issue by 0-wiz-0** [llvm/llvm-project#59978]
Building libunwind as a separate package for the pkgsrc software distribution, I noticed that (at least on NetBSD) the assembler files were not compiled at all.

`libunwind/src/CMakeLists.txt` has workarounds for macOS, mingw, and AIX where the assembler sources are defined to be 'C' source code. I guess this is working around the same issue, but I'm not sure.

Anyway, I filed a bug with CMake, but a CMake developer said [this was caused by libunwind not enabling assembler support](https://gitlab.kitware.com/cmake/cmake/-/issues/24308#note_1300126).
I tested the proposed fix, and it works for me.

Please add
```
enable_language(ASM)
```
at the top of libunwind/src/CMakeLists.txt, and find out if the following workaround:
```
# See add_asm_sources() in compiler-rt for explanation of this workaround.
# CMake doesn't work correctly with assembly on AIX. Workaround by compiling
# as C files as well.
if((APPLE AND CMAKE_VERSION VERSION_LESS 3.19) OR
   (MINGW AND CMAKE_VERSION VERSION_LESS 3.17) OR
   (${CMAKE_SYSTEM_NAME} MATCHES "AIX"))
  set_source_files_properties(${LIBUNWIND_ASM_SOURCES} PROPERTIES LANGUAGE C)
endif()
```
is really needed.

**Comment by compnerd** [comment]
Yes, adding `enable_language(ASM)` or adding `ASM` to the `cmake_project`'s `LANGUAGES` is the correct thing to do.  I would personally prefer the latter over the former.
